cargo-concordium/src/main.rs

```diff
@@ -334,6 +375,167 @@ pub fn main() -> anyhow::Result<()> {
+                                     smart contract module with an embedded module.",
+                                    err
+                                ))
+                            );
+                        }
+                        schema
+                    }
+                };
+
+                schema
+                    .as_ref()
+                    .map_err(|_| anyhow::anyhow!("Could not deserialize schema file."))?;
+
+                match schema.as_ref().unwrap() {
+                    VersionedModuleSchema::V0(module_schema) => {
+                        for (contract_name, contract_schema) in module_schema.contracts.iter() {
+                            write_json_schema_to_file_v0(
+                                absolute_path_of_out.clone(),
+                                contract_name,
+                                contract_schema,
+                            )?
+                        }
+                    }
+                    VersionedModuleSchema::V1(module_schema) => {
+                        for (contract_name, contract_schema) in module_schema.contracts.iter() {
+                            write_json_schema_to_file_v1(
+                                absolute_path_of_out.clone(),
+                                contract_name,
+                                contract_schema,
+                            )?
+                        }
+                    }
+                    VersionedModuleSchema::V2(module_schema) => {
+                        for (contract_name, contract_schema) in module_schema.contracts.iter() {
+                            write_json_schema_to_file_v2(
+                                absolute_path_of_out.clone(),
+                                contract_name,
+                                contract_schema,
+                            )?
+                        }
+                    }
+                    VersionedModuleSchema::V3(module_schema) => {
+                        for (contract_name, contract_schema) in module_schema.contracts.iter() {
+                            write_json_schema_to_file_v3(
+                                absolute_path_of_out.clone(),
+                                contract_name,
+                                contract_schema,
+                            )?
+                        }
+                    }
+                }
+            }
+
+            if let Some(schema_path) = schema_path {
+                let bytes = fs::read(schema_path).context("Could not read schema file.")?;
+
+                let schema = if bytes.starts_with(VERSIONED_SCHEMA_MAGIC_HASH) {
+                    from_bytes::<VersionedModuleSchema>(&bytes)
+                } else {
+                    // Question: What is if this is a schema originated from a `WasmVersion::V0`
+                    // module, then this should be 'VersionedModuleSchema::V0', isn't it?
+                    from_bytes(&bytes).map(VersionedModuleSchema::V1)
```

Comment:
Yes, you have to ask this from the user, as an extra option on the command-line.